python/ray/serve/tests/test_proxy_state.py

```diff
@@ -41,84 +39,92 @@ def get_alive_node_ids(self):
         return {node_id for node_id, _ in self.alive_nodes}
 
 
-def _make_proxy_state_manager(
-    http_options: HTTPOptions,
-    head_node_id: str = HEAD_NODE_ID,
-    cluster_node_info_cache=MockClusterNodeInfoCache(),
-    proxy_actor_class=ProxyActor,
-    actor_proxy_wrapper_class=ActorProxyWrapper,
-) -> (ProxyStateManager, ClusterNodeInfoCache):
-    return (
-        ProxyStateManager(
-            SERVE_CONTROLLER_NAME,
-            config=http_options,
-            head_node_id=head_node_id,
-            cluster_node_info_cache=cluster_node_info_cache,
-            proxy_actor_class=proxy_actor_class,
-            actor_proxy_wrapper_class=actor_proxy_wrapper_class,
-        ),
-        cluster_node_info_cache,
-    )
+class FakeProxyActor:
+    def __init__(self, *args, **kwargs):
+        pass
 
+    async def ready(self):
+        return json.dumps(["mock_worker_id", "mock_log_file_path"])
 
-@pytest.fixture
-def number_of_worker_nodes() -> int:
-    return 100
+    async def check_health(self):
+        pass
 
 
-@pytest.fixture
-def all_nodes(number_of_worker_nodes) -> List[Tuple[str, str]]:
-    return [(HEAD_NODE_ID, "fake-head-ip")] + [
-        (f"worker-node-id-{i}", f"fake-worker-ip-{i}")
-        for i in range(number_of_worker_nodes)
-    ]
+class FakeProxyWrapper(ProxyWrapper):
+    def __init__(self, actor_handle=None, *args, **kwargs):
+        self.actor_handle = actor_handle or FakeProxyActor(*args, **kwargs)
```

Comment:
why do we need to support a real actor_handle here? is this temporary as you continue refactoring tests?

Reply:
oops, it was for `test_proxy_state_update_healthy_check_health_sometimes_fails`. Testing the behavior where the health check is called x number of times in between proxy failing and succeeding health checks